File: task-03/Python/Subtask4.py
# Read the number from input.txt
with open("input.txt", "r") as file:
    n = int(file.read())

# Write the diamond pattern to output.txt
with open("output.txt", "w") as file:
    # Top half of the diamond
    for i in range(n):
        file.write(" " * (n - i - 1) + "*" * (2 * i + 1) + "\n")
    
    # Bottom half of the diamond
    for i in range(n-2, -1, -1):
        file.write(" " * (n - i - 1) + "*" * (2 * i + 1) + "\n")

File: task-03/Ruby/Subtask4.rb
# Read the number from input.txt
n = File.read("input.txt").to_i

# Write the diamond pattern to output.txt
File.open("output.txt", "w") do |file|
  # Top half of the diamond
  for i in 0...n
    file.puts " " * (n - i - 1) + "*" * (2 * i + 1)
  end

  # Bottom half of the diamond
  for i in (n-2).downto(0)
    file.puts " " * (n - i - 1) + "*" * (2 * i + 1)
  end
end

